test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -451,66 +458,71 @@ static void alg_test(odp_crypto_op_t op,
 	odp_packet_t pkt = odp_packet_alloc(suite_context.pool,
 					    ref->length + ref->digest_length);
 	CU_ASSERT(pkt != ODP_PACKET_INVALID);
-restart:
-	if (op == ODP_CRYPTO_OP_ENCODE) {
-		odp_packet_copy_from_mem(pkt, 0, ref->length, ref->plaintext);
-	} else {
-		odp_packet_copy_from_mem(pkt, 0, ref->length, ref->ciphertext);
-		odp_packet_copy_from_mem(pkt, ref->length,
-					 ref->digest_length,
-					 ref->digest);
-		if (should_fail) {
-			uint8_t byte = ~ref->digest[0];
 
+	for (iteration = NORMAL_TEST; iteration < MAX_TEST; iteration++) {
+		/* checking against wrong digest is meaningless for NULL digest
+		 * or when generating digest */
+		if (iteration == WRONG_DIGEST_TEST &&
+		    (auth_alg == ODP_AUTH_ALG_NULL ||
+		     op == ODP_CRYPTO_OP_ENCODE))
+			continue;
+
+		if (op == ODP_CRYPTO_OP_ENCODE) {
+			odp_packet_copy_from_mem(pkt, 0, ref->length,
+						 ref->plaintext);
+		} else {
```

Comment:
continue here will give you one tab room for rest of code. But it's up to you.

Reply:
@muvarov The loop doesn't end here since there's additional processing for both arms following the `else` clause.